### DISCUSSION TITLE
Pre-defined transform without matcher logic

### DISCUSSION THREAD

**Discussion by TrevorRice** [amzn/style-dictionary#1292] [Q&A] [answered]
Apologies if I missed this in the docs, but is there an easy way to leverage a pre-defined transform _without_ the built-in `matcher` logic?

For reasons I'm not privy to, our team opted to use `colors` instead of `color` when defining our tokens. This means, for instance, when running our tokens through `ios-swift-separate`, the [`color/UIColorSwift`](https://github.com/amzn/style-dictionary/blob/17f4cb2f30bd002dfd55d6ef8c5bee4138de8d64/lib/common/transforms.js#L465-L475) transform is not applied because [`isColor`](https://github.com/amzn/style-dictionary/blob/17f4cb2f30bd002dfd55d6ef8c5bee4138de8d64/lib/common/transforms.js#L20-L22) does not match our `colors` category.

I know I could rework/reorganize some things, but I'm hoping to avoid reimplementing `color/UIColorSwift` (and pulling in its dependency) simply to apply the transform to our `colors` category instead of `color`.

**Comment by jorenbroekema** [comment, marked as answer]
~~Hm no this is currently not possible, transforms aren't exposed publicly without the attached matcher/filter (called filter in v4).~~

~~I'd be open to making the transforms themselves available without the filters though, feel free to raise an issue and I'll mark it as enhancement~~

You can kinda do it in v4:

```js
const colorHslBuiltin = StyleDictionary.hooks.transforms['color/hsl'];
StyleDictionary.registerTransform({
  ...colorHslBuiltin,
  filter: (token) => token.type === 'colors',
  name: 'colors/hsl', // if you don't override the name it will register on the original name and override the built-in one, which might be nice
});
```

**Reply by TrevorRice** [reply]
Interesting! Thanks, I'll play around with this.
